frontend/src/features/public-form/components/FormStartPage/FormHeader.tsx

```diff
@@ -122,7 +175,9 @@ export const FormHeader = (): JSX.Element | null => {
 
   return (
     <>
-      <MiniHeader isOpen={isOpen} />
+      <FormSectionsProvider form={form}>
```

Comment:
should we move `FormSectionsProvider` out to the parent instead, or would that cause too much performance impact? 

Should experiment so we only instantiate a single  `FormSectionsProvider`, or we may encounter subtle bugs in the future (where state changes to a deeper nested provider will not affect the main provider that is subscribed by other components)

Reply:
I've moved `FormSectionsProvider` out to a new `PublicFormWithHeaderWrapper`, since both `FormHeader` and `FormFields` require it